realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1188,15 +1188,10 @@ public void run() {
                     transaction.execute(bgRealm);
 
                     if (!Thread.currentThread().isInterrupted()) {
-                        bgRealm.commitAsyncTransaction(new Runnable() {
-                            @Override
-                            public void run() {
-                                // The bgRealm needs to be closed before post event to caller's handler to avoid
-                                // concurrency problem. eg.: User wants to delete Realm in the callbacks.
-                                // This will close Realm before sending REALM_CHANGED.
-                                bgRealm.close();
-                            }
-                        });
+                        bgRealm.commitAsyncTransaction();
+                        // The bgRealm needs to be closed before post event to caller's handler to avoid concurrency
```

Comment:
The bgRealm needs to be closed before posting the REALM_CHANGED event to the caller's handler to avoid currency problems.his is currently guaranteed by posting handleAsyncTransactionCompleted below.